tests/test_cli.py

```diff
@@ -0,0 +1,343 @@
+            'standard_outputs.zip',
+            'link_simp_map.json'
+        ]
+    )
+
+
+def test_squeeze_external_area(invoke_runner_and_check_files):
+    invoke_runner_and_check_files(
+        "squeeze_external_area",
+        args= [
+            f'--network={EXAMPLE_NETWORK}',
+            f'--projection={PROJECTION}',
+            f'--study_area={os.path.join(EXAMPLE_DATA_DIR, "Fitzrovia_polygon.geojson")}',
+            '--freespeed=0.5',
+            '--capacity=0.5',
+
+        ],
+        expected_files=[
+            'network.xml',
+            os.path.join('supporting_outputs', 'external_network_links.geojson'),
+            os.path.join('supporting_outputs', 'capacity_before.geojson'),
+            os.path.join('supporting_outputs', 'freespeed_after.geojson'),
+            os.path.join('supporting_outputs', 'capacity_after.geojson'),
+            os.path.join('supporting_outputs', 'freespeed_before.geojson'),
+        ]
+    )
+
+
+def test_squeeze_urban_links(invoke_runner_and_check_files):
+    invoke_runner_and_check_files(
+        "squeeze_urban_links",
+        args= [
+            f'--network={EXAMPLE_NETWORK}',
+            f'--projection={PROJECTION}',
+            f'--study_area={os.path.join(EXAMPLE_DATA_DIR, "Fitzrovia_polygon.geojson")}',
+            f'--urban_geometries={os.path.join(EXAMPLE_DATA_DIR, "Fitzrovia_urban_polygon.geojson")}',
+            '--freespeed', '0.5',
+            '--capacity', '0.5',
+
+        ],
+        expected_files=[
+            'network.xml',
+            os.path.join('supporting_outputs', 'capacity_before.geojson'),
+            os.path.join('supporting_outputs', 'freespeed_after.geojson'),
+            os.path.join('supporting_outputs', 'urban_network_links.geojson'),
+            os.path.join('supporting_outputs', 'capacity_after.geojson'),
+            os.path.join('supporting_outputs', 'freespeed_before.geojson')
+        ]
+    )
+
+
+def test_validate_network(invoke_runner_and_check_files):
+    invoke_runner_and_check_files(
+        "validate_network",
+        args= [
+            f'--network={EXAMPLE_NETWORK}',
+            f'--schedule={EXAMPLE_SCHEDULE}',
+            f'--vehicles={EXAMPLE_VEHICLES}',
+            f'--projection={PROJECTION}',
+        ],
+        expected_files=[
+            'validation_report.json'
+        ]
+    )
+
+
+def test_google_api_script_throws_error_without_api_key(invoke_runner_and_check_files):
+    with pytest.raises(AssertionError) as excinfo:
+        invoke_runner_and_check_files(
+            "send_google_directions_requests_for_network",
+            args= [
+                f'--network={EXAMPLE_NETWORK}',
+                f'--projection={PROJECTION}',
+                '--requests_threshold=1',
+                '--subset_conditions=primary,motorway',
+                '--traffic_model=best_guess',
+                '--departure_time=now'
+            ],
+            not_expected_files=[
+                'api_requests.json',
```

Comment:
the expectation is that the requests will be generated and saved even if sending them fails